internal/provider/dedicated_cluster_resource.go

```diff
@@ -0,0 +1,792 @@
+	Annotations        types.Map           `tfsdk:"annotations"`
+	TiDBNodeSetting    tidbNodeSetting     `tfsdk:"tidb_node_setting"`
+	TiKVNodeSetting    tikvNodeSetting     `tfsdk:"tikv_node_setting"`
+	TiFlashNodeSetting *tiflashNodeSetting `tfsdk:"tiflash_node_setting"`
+}
+
+type pausePlan struct {
+	PauseType           types.String `tfsdk:"pause_type"`
+	scheduledResumeTime types.String `tfsdk:"scheduled_resume_time"`
+}
+
+type tidbNodeSetting struct {
+	NodeSpecKey          types.String `tfsdk:"node_spec_key"`
+	NodeCount            types.Int32  `tfsdk:"node_count"`
+	NodeGroupId          types.String `tfsdk:"node_group_id"`
+	NodeGroupDisplayName types.String `tfsdk:"node_group_display_name"`
+	NodeSpecDisplayName  types.String `tfsdk:"node_spec_display_name"`
+	IsDefaultGroup       types.Bool   `tfsdk:"is_default_group"`
+	State                types.String `tfsdk:"state"`
+	Endpoints            []endpoint   `tfsdk:"endpoints"`
+}
+
+type endpoint struct {
+	Host           types.String `tfsdk:"host"`
+	Port           types.Int32  `tfsdk:"port"`
+	ConnectionType types.String `tfsdk:"connection_type"`
+}
+
+type tikvNodeSetting struct {
+	NodeSpecKey         types.String `tfsdk:"node_spec_key"`
+	NodeCount           types.Int32  `tfsdk:"node_count"`
+	StorageSizeGi       types.Int32  `tfsdk:"storage_size_gi"`
+	StorageType         types.String `tfsdk:"storage_type"`
+	NodeSpecDisplayName types.String `tfsdk:"node_spec_display_name"`
+}
+
+type tiflashNodeSetting struct {
+	NodeSpecKey         types.String `tfsdk:"node_spec_key"`
+	NodeCount           types.Int32  `tfsdk:"node_count"`
+	StorageSizeGi       types.Int32  `tfsdk:"storage_size_gi"`
+	StorageType         types.String `tfsdk:"storage_type"`
+	NodeSpecDisplayName types.String `tfsdk:"node_spec_display_name"`
+}
+
+type dedicatedClusterResource struct {
+	provider *tidbcloudProvider
+}
+
+func NewDedicatedClusterResource() resource.Resource {
+	return &dedicatedClusterResource{}
+}
+
+func (r *dedicatedClusterResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dedicated_cluster"
+}
+
+func (r *dedicatedClusterResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	var ok bool
+	if r.provider, ok = req.ProviderData.(*tidbcloudProvider); !ok {
+		resp.Diagnostics.AddError("Internal provider error",
+			fmt.Sprintf("Error in Configure: expected %T but got %T", tidbcloudProvider{}, req.ProviderData))
+	}
+}
+
+func (r *dedicatedClusterResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "dedicated cluster resource",
+		Attributes: map[string]schema.Attribute{
+			"project_id": schema.StringAttribute{
```

Comment:
I think project_id should be optional to let users define which project want to use